filecoin-proofs/src/api/seal.rs

```diff
@@ -758,3 +759,64 @@ pub fn verify_batch_seal<Tree: 'static + MerkleTreeTrait>(
     info!("verify_batch_seal:finish");
     result
 }
+
+pub fn fauxrep<R: AsRef<Path>, S: AsRef<Path>, Tree: 'static + MerkleTreeTrait>(
+    porep_config: PoRepConfig,
+    cache_path: R,
+    out_path: S,
+) -> Result<Commitment> {
+    let mut rng = rand::thread_rng();
+    fauxrep_aux::<_, R, S, Tree>(&mut rng, porep_config, cache_path, out_path)
+}
+
+pub fn fauxrep_aux<
+    Rng: rand::Rng,
+    R: AsRef<Path>,
+    S: AsRef<Path>,
+    Tree: 'static + MerkleTreeTrait,
+>(
+    mut rng: &mut Rng,
+    porep_config: PoRepConfig,
+    cache_path: R,
+    out_path: S,
+) -> Result<Commitment> {
+    let sector_bytes = PaddedBytesAmount::from(porep_config).0 as usize;
+
+    {
+        // Create a sector full of null bytes at `out_path`.
```

Comment:
I think we can use this instead of allocating/looping: https://doc.rust-lang.org/std/fs/struct.File.html#method.set_len

Reply:
Good call. I pushed that fix and also one to reuse the vector when building on GPU.